SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/ProcessRunner.swift

```diff
@@ -12,6 +12,29 @@
 
 import Foundation
 
+/// Keeps track of subprocesses spawned by this script and forwards SIGINT
+/// signals to them.
+class SigIntListener {
+  /// The subprocesses spawned by this script that are currently running.
+  static var runningSubprocesses: Set<Process> = []
+
+  /// Whether a `SIGINT` signal has been received by this script.
+  static var hasReceivedSigInt: Bool = false
+
+  /// Registers a `SIGINT` signal handler that forwards `SIGINT` to all
+  /// subprocesses that are registered in `runningSubprocesses`
+  static func registerSigIntSubprocessTerminationHandler() {
+    #if canImport(Darwin) || canImport(Glibc)
+    signal(SIGINT) { _ in
+      SigIntListener.hasReceivedSigInt = true
+      for process in SigIntListener.runningSubprocesses {
+        kill(process.processIdentifier, SIGINT)
```

Comment:
There's a `Process.terminate` and `Process.interrupt` you could use instead signalling directly

Reply:
Oh, nice. I didn’t know about that.